### PR TITLE
add docker-compose template. add `Port` config. add extra-options

### PR DESCRIPTION
**Pull request by yorkane**
Fix my.cnf  mapping cause mysql start with `port:0` (introduced by mariadb-server.cnf)
Deletions information output
Draft template for docker-compose.yml
Make `Port` `mysqld-commandline-options` configurable

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,6 +35,7 @@ RUN \
     tr -d " \t\n\r" | sed -e 's/usr/|usr/g' -e 's/^.//') && \
   INSTALLED=$(apk info -q -L mariadb-common mariadb linux-pam | grep "\S") && \
   for path in $(echo "${INSTALLED}" | grep -v -E "${TO_KEEP}"); do \
+    echo "remove ${path}";\
```
Comment (jbergstroem) [inline Dockerfile:38]: The changes to the `Dockerfile` looks unrelated to the compose introduction. Are they needed?

File: Dockerfile [hunk 2]
```diff
@@ -43,7 +44,8 @@ RUN \
   # allow anyone to connect by default
   sed -ie 's/127.0.0.1/%/' /usr/share/mariadb/mysql_system_tables_data.sql && \
   mkdir /run/mysqld && \
-  chown mysql:mysql /etc/my.cnf.d/ /run/mysqld /usr/share/mariadb/mysql_system_tables_data.sql
+  chown mysql:mysql /etc/my.cnf.d/ /run/mysqld /usr/share/mariadb/mysql_system_tables_data.sql && \
+  rm /etc/my.cnf.d/* -rf
```
Comment (jbergstroem) [inline Dockerfile:48]: Is it necessary to remove this content? The idea is that you lean on mariadb defaults or mount your own config folder. I can see how it could be problematic if both configs are loaded; is that perhaps whats happening for you?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+version: "3"
+services:
+  mysql:
+    image: jbergstroem/mariadb-alpine
+    container_name: mysql
+    ports:
+      - "3306:3306/tcp"
+    volumes:
+      - ${DATA_ROOT:-.}/db/data:/var/lib/mysql # where database and logs preserve
```
Comment (jbergstroem) [inline docker-compose.yml:9]: You should create volumes here instead of mounting a path

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+version: "3"
+services:
+  mysql:
```
Comment (jbergstroem) [inline docker-compose.yml:3]:
```suggestion
  mariadb:
```

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    image: jbergstroem/mariadb-alpine
+    container_name: mysql
```
Comment (jbergstroem) [inline docker-compose.yml:5]:
```suggestion
    container_name: mariadb
```

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    ports:
+      - "3306:3306/tcp"
+    volumes:
+      - ${DATA_ROOT:-.}/db/data:/var/lib/mysql # where database and logs preserve
+      - ${DATA_ROOT:-.}/db/init:/docker-entrypoint-initdb.d # All xxx.sql, xxx.sql.gz will load into MY_DB. All `xxx.sh` will be executed
+    environment:
+      - MYSQL_DATABASE=${MY_DB:-mtest}
+      - MYSQL_CHARSET=${MY_DB_CHARSET:-utf8mb4}
```
Comment (jbergstroem) [inline docker-compose.yml:13]: I believe mariadb already has a set of default; is there a point to overriding their defaults with these ones?

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      - MYSQL_COLLATION=${MY_DB_COLLATION:-utf8mb4_general_ci}
+      - MYSQL_USER=${MY_DB_USER:-mtest}
+      - MYSQL_PASSWORD=${MY_DB_PASSWORD:-mtest} #
+      - MYSQL_ROOT_PASSWORD=${MY_ROOT_PASSWORD:-root} # set the root password
```
Comment (jbergstroem) [inline docker-compose.yml:17]: This would change the default behavior of how the container works today; I think we should assume as few defaults as possible.

File: sh/run.sh [hunk 8]
```diff
@@ -14,7 +14,7 @@ if [ -z "$(ls -A /etc/my.cnf.d/* 2> /dev/null)" ]; then
         -e '/^innodb/d' /etc/my.cnf.d/my.cnf
 fi
 
-MYSQLD_OPTS="--user=mysql"
+MYSQLD_OPTS="--user=mysql --port=${MYSQL_PORT} ${MYSQL_EXT_OPTS}"
```
Comment (jbergstroem) [inline sh/run.sh:17]: My idea here was to override the port via a config instead of having to modify the shell script. Does that make sense?

**Review by jbergstroem — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (yorkane) [reply Dockerfile:38]: I was failed to upgrade mariadb version. I think it's necessary to know which files was deleted during the Docker-image-building process.

File: Dockerfile (hunk 2 above)
Comment (yorkane) [reply Dockerfile:48]: #63 showed in my case.  I found both original `mariadb-server.cnf`  and `my.cnf` (with copied from `/tmp/my.cnf`). Then mysqld started with `port  0`
seems the default `cnf-file` will impact the initialization process.

File: docker-compose.yml (hunk 6 above)
Comment (yorkane) [reply docker-compose.yml:13]: Just port the environment-variable setting from your Dockerfile, and leave default values.

File: docker-compose.yml (hunk 7 above)
Comment (yorkane) [reply docker-compose.yml:17]: Just port the environment-variable setting from your Dockerfile, and leave default values.

File: sh/run.sh (hunk 8 above)
Comment (yorkane) [reply sh/run.sh:17]: Some cases would create mutiple mariadb instances within the same box. A configurable port option would be necessary.
And with your design, leave port config in environment-variable is more conveniant. since the mariadb is *simplified* .

**Comment by jbergstroem** [comment]
I will add a compose example in `docs/usage.md` based on your PR! Thanks a lot for your contribution.
